lib/SIL/SILPrinter.cpp

```diff
@@ -2787,6 +2818,8 @@ void SILModule::print(SILPrintContext &PrintCtx, ModuleDecl *M,
   }
 
   printSILGlobals(PrintCtx, getSILGlobalList());
+  printSILDifferentiabilityWitnesses(PrintCtx,
```

Comment:
Note: printing SIL differentiability witnesses before SIL functions may be slightly controversial.

This decision significantly simplifies SIL parsing: it allows us to avoid adding support for diff. witness forward declarations when parsing `differentiability_witness_function` instructions in SIL functions.

Other longer table-like SIL constructs are printed after SIL functions so that functions appear at the top. Diff. witnesses aren't as long as other table-like SIL constructs (up to four lines), so they don't take up too much space.